transport_nantes/mobilito/templates/mobilito/mobilito_session_summary.html

```diff
@@ -4,118 +4,142 @@
-                            </h5>
-                            <button type="button" class="close" data-dismiss="modal" aria-label="Close">
-                                <span aria-hidden="true">&times;</span>
-                            </button>
-                        </div>
-                        <div class="modal-body">
-                            <form id="report-abuse-form" method="post">
-                                {% csrf_token %}
-                                <input type="hidden"
-                                    name="mobilito_session_id"
-                                    value="{{ mobilito_session.id }}"/>
-                                <div class="form-group">
-                                    <label for="report-abuse-text">
-                                        Raison du signalement
-                                    </label>
-                                    <textarea
-                                    class="form-control"
-                                    id="report-abuse-text"
-                                    name="report-abuse-text"
-                                    rows="3"></textarea>
-                                </div>
-                                <button type="submit"
-                                    class="btn donation-button">
-                                    Signaler
-                                </button>
-                            </form>
-                        </div>
-                    </div>
-                </div>
-            </div>
-        {% endif %}
-        {% endif %}
-        
-        
-    </article>
-{% endblock app_content %}
\ No newline at end of file
+{% if request.user != mobilito_session.user.user %}
+{% if not user_has_reported_this_session %}
+    <div id="report-abuse-div"
+    class="modal fade"
+    tabindex="-1"
+    aria-labelledby="report-about-form-title"
+    aria-hidden="true">
+	<div class="modal-dialog modal-dialog-centered">
+	    <div class="modal-content">
+		<div class="modal-header">
+		    <h5 class="modal-title" id="report-about-form-title">
+			Signaler une session mobilito
+		    </h5>
+		    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+			<span aria-hidden="true">&times;</span>
+		    </button>
+		</div>
+		<div class="modal-body">
+		    <form id="report-abuse-form" method="post">
+			{% csrf_token %}
+			<input type="hidden"
+			    name="mobilito_session_id"
+			    value="{{ mobilito_session.id }}"/>
+			<div class="form-group">
+			    <label for="report-abuse-text">
+				Raison du signalement
+			    </label>
+			    <textarea
+			    class="form-control"
+			    id="report-abuse-text"
+			    name="report-abuse-text"
+			    rows="3"></textarea>
+			</div>
+			<button type="submit"
+			    class="btn donation-button">
+			    Signaler
+			</button>
+		    </form>
+		</div>
+	    </div>
+	</div>
+    </div>
+{% endif %}
+{% endif %}
+{% endblock app_content %}
```

Comment:
Fixed some unclosed tags and added indent, without changing its structure otherwise (will wait for the error to be fixed before seeing to change layout)

```suggestion
{% block styles %}
	{{block.super}}
	<link rel="stylesheet" href="{% static 'mobilito/css/session_summary.css' %}">
{% endblock styles %}

{% block scripts %}
	{{ block.super }}
	{% compress js %}
		<script
			src="{% static 'mobilito/js/report_mobilito_session.js' %}"
			defer>
		</script>
	{% endcompress %}
{% endblock scripts %}


{% block app_content %}
<div class="container">
    <div class="row">
		<div class="col-lg-12">
			{# name of recorder, some day #}
			<p>{{ mobilito_session.start_timestamp|timesince:mobilito_session.end_timestamp }}
				à partir de {{ mobilito_session.start_timestamp|timezone:'Europe/Paris'|date:"l j F Y à G:i" }}.
			</p>
			<p><b>{{ mobilito_session.location }}</b></p>
			<p>[carte ici]</p>

			<p>
				<img src="{% url 'mobilito:mobilito_session_timeseries_image' mobilito_session.session_sha1 %}"
				alt="{{ mobilito_session.location }}"
				width="90%">
			</p>
		</div>
    </div>
    <div class="row">
		<div class="col-xs-3">
			<p>
				<img src="{% url 'mobilito:mobilito_session_fraction_image' mobilito_session.session_sha1 %}"
				alt="{{ mobilito_session.location }}"
				width="90%">
			</p>
		</div>
		<div class="col-xs-9">
			<p>
				<ul>
					<li> {{ mobilito_session.pedestrian_count|default:"0" }} piétons</li>
					<li> {{ mobilito_session.bicycle_count|default:"0" }} vélos</li>
					<li> {{ mobilito_session.motor_vehicle_count|default:"0" }} voitures et camions</li>
					<li> {{ mobilito_session.public_transport_count|default:"0" }}
					transports collectifs</li>
				</ul>
			</p>
		</div>
    </div>
</div>

<div class="container">
    <div class="row">
		<div class="col-lg-12">
			<div class="dropdown show">
				<a href="#" role="button" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
					<i class="fa-solid fa-flag ml-md-3"></i>
				</a>
				<div class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuLink">
					<a class="dropdown-item" href="#"
					{% if user_has_reported_this_session %}style="pointer-events: none; color: grey;"{% endif %}>
					{% if request.user != mobilito_session.user.user %}
						{% if user_has_reported_this_session %}
						<span id="reported">
							Session signalée
						</span>
						{% else%}
						<span id="reported" class="d-none"
						style="color: grey;">
							Session signalée
						</span>
						<span id="report-abuse"
						data-toggle="modal" data-target="#report-abuse-div">
							Signaler
						</span>
						{% endif %}
					{% endif %}
					</a>
				</div>
			</div>
		</div>
    </div>
</div>


{% if request.user != mobilito_session.user.user %}
	{% if not user_has_reported_this_session %}
		<div id="report-abuse-div"
		class="modal fade"
		tabindex="-1"
		aria-labelledby="report-about-form-title"
		aria-hidden="true">
			<div class="modal-dialog modal-dialog-centered">
				<div class="modal-content">
					<div class="modal-header">
						<h5 class="modal-title" id="report-about-form-title">
						Signaler une session mobilito
						</h5>
						<button type="button" class="close" data-dismiss="modal" aria-label="Close">
						<span aria-hidden="true">&times;</span>
						</button>
					</div>
					<div class="modal-body">
						<form id="report-abuse-form" method="post">
							{% csrf_token %}
							<input type="hidden"
								name="mobilito_session_id"
								value="{{ mobilito_session.id }}"/>
							<div class="form-group">
								<label for="report-abuse-text">
									Raison du signalement
								</label>
								<textarea
								class="form-control"
								id="report-abuse-text"
								name="report-abuse-text"
								rows="3"></textarea>
							</div>
							<button type="submit"
								class="btn donation-button">
								Signaler
							</button>
						</form>
					</div>
				</div>
			</div>
		</div>
	{% endif %}
{% endif %}

{% endblock app_content %}

```

Reply:
That's so readable said like that!  🤣 

But, thanks!  I'll improve it.

I looked at your improved branch.  One minor issue is that you don't preserve the roundness of the pie chart.

![image](https://user-images.githubusercontent.com/1220683/202922448-3c04b2bc-01ac-4788-9891-e8ad90c06c8a.png)

But I'll take this on and sort it, I need to understand this sort of thing properly.

Reply:
Haha yeah, I tried the `suggestion` block in github's review markdown, but that block is very large so it's not very readable ! 

On this CSS issue, it looks like the display is forced in a flex-direction: row that doesn't wrap.
I think the image should have some responsive width with the col-`breakpoint`-<1-12>
Remember the way it works : 
col-12 will set width to 100 
if you have col-12 (which is equivalent to col-xs-12) and another class like col-md-8, it means that from 0px to the md breakpoint at 767px it applies col-12, and from 768 to infinite, it's col-md-8 that is applied.

https://getbootstrap.com/docs/4.6/layout/overview/